src/wasm-builder.h

```diff
@@ -321,15 +321,17 @@ class Builder {
 
   static Index addVar(Function* func, Name name, WasmType type) {
     // always ok to add a var, it does not affect other indices
+    assert(name.is());
     Index index = func->getNumLocals();
-    if (name.is()) func->localIndices[name] = index;
+    func->localIndices[name] = index;
     func->localNames.push_back(name);
     func->vars.emplace_back(type);
     return index;
   }
 
   static Index addVar(Function* func, WasmType type) {
-    return addVar(func, Name(), type);
+    Name name = func->getLocalNameOrGeneric(func->getNumLocals());
```

Comment:
Why even call that method? it will not find the local so it just returns `Name::fromInt`, I think?

We should also guard against colliding with another name, as in theory a previous variable could have a name which is a larger integer. So we could loop on a counter until we find a free one (which should almost never hit the loop).